### PR TITLE
[api] fix method getTransactionCount

### LINKED ISSUE #2912: GetTransactionCount returns the number of transactions the given address has sen...
https://github.com/iotexproject/iotex-core/blob/fd68a083e4fbaa1994647b0b40c9c1bcf4d2d3c8/api/web3server.go#L343-L348

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in fd68a083e4fbaa1994647b0b40c9c1bcf4d2d3c8. It&#x27;s been assigned to @Liuhaai because they committed the code.

### PR DESCRIPTION
**Pull request by Liuhaai**
fix #2912

### CODE REVIEW

File: api/web3server.go [hunk 1]
```diff
@@ -367,11 +366,33 @@ func (svr *Web3Server) getTransactionCount(in interface{}) (interface{}, error)
...
 	if err != nil {
 		return nil, err
 	}
-	return uint64ToHex(accountMeta.PendingNonce), nil
+	blkMetas, err := svr.coreService.BlockMetas(num, 1)
+	if err != nil || blkMetas == nil || len(blkMetas) == 0 {
+		if err == nil {
+			err = errors.New("invalid block")
+		}
+		return nil, err
```
Comment (CoderZhi) [inline api/web3server.go:382]: break into two if/return

**Review by CoderZhi — APPROVED** [review]

File: api/web3server.go [hunk 2]
```diff
@@ -367,11 +367,33 @@ func (svr *Web3Server) getTransactionCount(in interface{}) (interface{}, error)
...
+	if err != nil {
+		return nil, err
+	}
+	blkMetas, err := svr.coreService.BlockMetas(num, 1)
 	if err != nil {
 		return nil, err
 	}
-	return uint64ToHex(accountMeta.PendingNonce), nil
+	if len(blkMetas) == 0 {
+		return nil, errors.New("invalid block")
```
Comment (dustinxie) [inline api/web3server.go:383]: define an `errInvalidArgument` and return it?

File: api/web3server.go [hunk 3]
```diff
@@ -678,12 +700,12 @@ func (svr *Web3Server) getBlockTransactionCountByNumber(in interface{}) (interfa
...
 	blkMetas, err := svr.coreService.BlockMetas(num, 1)
-	if err != nil || blkMetas == nil || len(blkMetas) == 0 {
-		if err == nil {
-			err = errors.New("invalid block")
-		}
+	if err != nil {
 		return nil, err
 	}
+	if len(blkMetas) == 0 {
+		return nil, errors.New("invalid block")
```
Comment (dustinxie) [inline api/web3server.go:707]: same

File: api/web3server.go [hunk 4]
```diff
@@ -726,12 +748,12 @@ func (svr *Web3Server) getTransactionByBlockNumberAndIndex(in interface{}) (inte
...
```
Comment (dustinxie) [inline api/web3server.go:755]: same

File: api/web3server.go [hunk 5]
```diff
@@ -332,12 +332,13 @@ func (svr *Web3Server) getBlockByNumber(in interface{}) (interface{}, error) {
...
```
Comment (dustinxie) [inline api/web3server.go:339]: same

**Review by dustinxie — APPROVED** [review]

**Pull request merged**